### ISSUE TITLE
GTA.Native.?A0xe506951e.NativeTask.Run() Crash

### ISSUE THREAD

**Issue by Dilapidated** [crosire/scripthookvdotnet#546]
On the script we've been developing we seem to be getting random crashes. Nothing shows specifically in the SHV or SHVDN logs but in the windows event viewer the error shows the following:

`Application: GTA5.exe
Framework Version: v4.0.30319
Description: The process was terminated due to an unhandled exception.
Exception Info: System.AccessViolationException
   at nativeCall()
   at GTA.Native.?A0xe506951e.NativeTask.Run()
   at GTA.ScriptDomain.DoTick()
   at GTA.ScriptDomain.DoTick()
   at <Module>.ManagedTick()`

Looking over the crash dump it appears to be from some function called NtWaitForMultipleObjects.

Does anyone know what may be the cause of this?

**Comment by zorg93** [comment]
Sounds like an issue with a native you were calling. can you post code from the script which caused the error

**Comment by Dilapidated** [comment]
I doubt the source would help because the code itself is rather large about 5,000 lines and also because there isn't a specific native that stands out. We have logging set with the script and show no errors in our logs, SHV logs, or SHVDN logs. Only that event log above is consistent. As far as crashing goes there isn't a specific thing that happens before crashing either. It could happen in 5 mins or in 2 hrs etc etc and is completely random. 

We do have quite a few wait() timers. Is it possible that could be related to such an error?  I got to be honest, I've never heard of NtWaitForMultipleObjects before now and know of nothing that native above references/relates too specifically.

**Comment by crosire** [comment]
If a native script function is used in an unsafe way (direct memory pointers that may not stay valid and such), SHVDN cannot guarantee any crash safety. If you use features from the `GTA.Native` namespace directly, I suggest start looking there, as the SHVDN wrappers should be safe to use.

**Comment by Dilapidated** [comment]
Thanks for the reply. We will look it over to see if any stand out. Crosire if possible can I email you the source so you can see with us? I'd appreciate that a lot if you're okay with it, I just don't want to put it here in public view if I can help it.

**Comment by crosire** [comment]
You may. But I cannot guarantee I find the time to look through it. I'm not on a workstation with GTA 5 installed in the moment.

**Comment by Dilapidated** [comment]
I understand and thank you, what would be a good email to reach you at?

**Comment by crosire** [comment]
crosiredev@gmail.com

**Comment by Dilapidated** [comment]
You must of added it shortly after I sent this hehe. We actually found the error crosire and no longer need assistance. It was a missing argument in a native call.

To find it we made a function wrapper that we pipe all our direct function calls through:

private void CallNative(Hash Native, int values)
{
writelog("NativeCaller",native.ToString());
Function.Call(Native, values);
}

Thanks for the assistance and reaching out though. I appreciate it a lot :)
